avalee0215 35 create a basic homepage draft

avalee0215 opened the pull request:
closes #35

## 🌟 Context <!-- What is the purpose of this PR? -->

This PR adds a basic draft of the AUCC homepage as requested in the ticket
It is just providing the minimal structure

---

## 📝 Description <!--What changes have been made? -->

- Created `RecentContent.tsx` as a server component to fetch and render:
  - The 3 most recent published Events
  - The 3 most recent published Trip Reports
- Displayed event and trip titles with created dates using basic HTML (`<ul>`, `<li>`)
- Added `loading.tsx` for the page-level loading
- Implemented basic error handling with a fallback `<p>` message

---

## 📋 Notes <!--Are there any important details for reviewers? -->

<!--
Potential risks or breaking changes.
Edge cases or areas requiring special attention.
Anything you want feedback on!
List any manual testing you've performed
Mention any documentation that needs updating as a result
-->

dyzhuu commented on src/components/RecentContent.tsx:
```diff
@@ -0,0 +1,62 @@
+import { Event, TripReport } from '../payload-types'
+
+export default async function RecentContent() {
+  const payload = await getPayload({ config })
```
you can use the cached getPayloadClient function in lib!

dyzhuu commented on src/components/RecentContent.tsx:
```diff
@@ -0,0 +1,62 @@
+export default async function RecentContent() {
+  const payload = await getPayload({ config })
+
+  let events: Event[] = []
+  let trips: TripReport[] = []
+  let error: string | null = null
```
prefer to use `const` instead of `let`. Despite the name, `const` arrays are actually still mutable

dyzhuu commented on src/components/RecentContent.tsx:
```diff
@@ -0,0 +1,62 @@
+  let trips: TripReport[] = []
+  let error: string | null = null
+
+  try {
+    const eventsRes = await payload.find({
+      collection: 'events',
+      limit: 3,
+      sort: '-createdAt',
+    })
+
+    const tripsRes = await payload.find({
+      collection: 'trip-reports',
+      limit: 3,
+      sort: '-createdAt',
+    })
+
+    events = eventsRes.docs
+    trips = tripsRes.docs
+  } catch (err) {
+    console.error(err)
+    error = 'Failed to load content.'
+  }
+
+  if (error) {
+    return <p style={{ color: 'red' }}>{error}</p>
+  }
```
also, we would prefer to define a query function for this where possible under src/queries rather than having the raw queries in the component itself. Do you think you can re-use any of the pre-defined queries for this?

dyzhuu commented on src/components/RecentContent.tsx:
```diff
@@ -0,0 +1,62 @@
+  }
+
+  if (error) {
+    return <p style={{ color: 'red' }}>{error}</p>
```
we are using [tailwind css](https://tailwindcss.com/docs/color) for styling

dyzhuu commented on src/components/RecentContent.tsx:
```diff
@@ -0,0 +1,62 @@
+    trips = tripsRes.docs
+  } catch (err) {
+    console.error(err)
+    error = 'Failed to load content.'
```
if this is the only error, you can just hard code the message

dyzhuu replied:
re: my other comment about separating this into a separate query function -- then we can have the try catch logic in there instead. and perhaps instead of throwing we can return `null`

dyzhuu reviewed (changes requested):
Nice! Just left some comments -- some comments may conflict with others, e.g. I said to move to use a server function which may make some of these comments redundant, but they are just good fyi's B)

dyzhuu commented on src/queries/tripReports.ts:
Oh, by my other comment, I meant to have a look at the `getTripReports` function -- the `limit` argument specifies how many results you want to return. Therefore, if you want to get the 3 most recent published trip reports, you can just call it like `getTripReports({ limit = 3 })`.

Since we want to limit this just to published events, you may want to introduce a new `where` field as well

dyzhuu commented on src/app/(frontend)/page.tsx:
```diff
@@ -1,9 +1,10 @@
 import Image from 'next/image'
 import { getPayload } from 'payload'
 
+import RecentContent from '@/components/RecentContent'
```
For component imports, can we use named imports?

```suggestion
import { RecentContent } from '@/components/RecentContent'
```

for components, can we use named imports? (and default imports for page components).

dyzhuu commented on src/components/RecentContent.tsx:
For components that are related to a specific page, the pattern for this repo will be to locate them in the `_components` folder under the same directory as the `page.tsx` file.

Since `RecentContent` is a component made just for home, and the route is `app/(frontend)/page.tsx`, you can move it to `app/(frontend)/_components/RecentContent.tsx`

We wil use the `@/components` folder for shared global components

dyzhuu reviewed (approved):
Looks good! I left a few small comments, but will tentatively approve the PR :)

phawatboom reviewed (approved):
lgtm!

dyzhuu commented:
hey @avalee0215, if this is done feel free to merge it in! looks like there are just a few conflicts

avalee0215 commented:
> hey @avalee0215, if this is done feel free to merge it in! looks like there are just a few conflicts
Hey @dyzhuu, oh, sorry. I mistakenly thought the PR was complete earlier because the 'resolve conflicts"'button was disabled...
Since the main branch has changed quite a bit while I was catching up, I made a few updates to align with it.

Based on your feedback, I’ve now updated RecentContent to be a homepage only component and added the where clause to queries/ events and trip reports

Let me know if there’s anything else to change.  
Thanks again!

avalee0215 commented:
> hey @avalee0215, if this is done feel free to merge it in! looks like there are just a few conflicts

Hey @dyzhuu, oh, sorry. I mistakenly thought the PR was complete earlier because the 'resolve conflicts"'button was disabled...
Since the main branch has changed quite a bit while I was catching up, I made a few updates to align with it.

Based on your feedback, I’ve now updated RecentContent to be a homepage only component and added the where clause to queries/ events and trip reports

Thanks again!
